packages/firestore/src/api/user_data_converter.ts

```diff
@@ -530,51 +530,60 @@ export class UserDataConverter {
     } else if (value instanceof DocumentKeyReference) {
       return new RefValue(value.databaseId, value.key);
     } else if (value instanceof FieldValueImpl) {
```

Comment:
Just a suggestion - we could move this block up to `parseData`. Then we would have a single place in the code that decides between our parseX implementations.

Reply:
Good idea, done, though that led to a bit more refactoring to uniformly handle context.path being null (sorry).